demes/demes.py

```diff
@@ -1997,6 +1997,36 @@ def in_generations(self) -> Graph:
         graph.generation_time = 1
         return graph
 
+    def rename_demes(self, names: MutableMapping[str, str]) -> Graph:
+        """
+        Rename demes according to a dictionary.
+
+        :param dict names:
+            A dictionary with deme names and new names.
+        :return:
+            A demographic model with renamed demes.
+        :rtype: Graph
+        """
+        graph = copy.deepcopy(self)
+        if not isinstance(names, MutableMapping):
```

Comment:
I don't think its necessary to do explicit type checking here. If a non-dict is passed, this function will fail anyhow with an indexing error of some sort.